### ISSUE TITLE
This library is incompatible with Vite

### ISSUE THREAD

**Issue by bmrejen** [linx4200/vue-color#248]
Publishing a .vue file worked with Webpack but this won't work anymore when using Vite. Vite can convert .vue files in js in development but it seems that the current .vue file won't be compatible with that transformation.

![image](https://user-images.githubusercontent.com/20118433/168183204-76e060e0-413e-423f-9ec7-d88ada591b1f.png)

**Comment by zealot128** [comment]
Using it like this with Vite/Rollup (tested with Vite 4.3):

- 1. need a newer tinycolor2 library than required by package.json, so hardcode the version into resolutions in package.json:

```json
{
  "dependencies": { 
     ...
      "vue-color": "^2.8.1",
      "tinycolor2": "^1.6.0",
   },
   "resolutions": {
      "tinycolor2": "1.6.0",
   }
}
```

And import as ESM:

```javascript
import { Chrome } from 'vue-color'
```

But it will still import as 100kb blob #203  so better to import the file, which should work:

```javascript
import Chrome from 'vue-color/src/components/Chrome.vue'
```

**Comment by zealot128** [comment]
@linx4200 yes, but we do not use Vue 3.0 but Vue 2.7 still - because of a huge number of components and dependency on UI libraries.
With the workaround from me above, it also works without any problems on Vue 2.7, just by importing the Vue File from src instead of using the precompiled stuff in dist.

**Comment by linx4200** [comment]
Hi @zealot128  After testing with `vite@6.2.0` + `vue@2.7.16` +  `vue-color@2.8.1`, I found that `vue-color@2.x` is compatible with Vite — no errors occurred during either development or build processes.

```vue
<template>
  <div>
    <photoshop-picker v-model="colors" />
  </div>
</template>

<script>
import { Photoshop } from 'vue-color';
export default {
  data () {
    return {
      colors: '#247'
    }
  },
  components: {
    'photoshop-picker': Photoshop
  }
}
</script>

```

**Comment by linx4200** [comment]
As mentioned in https://github.com/linx4200/vue-color/issues/275, I’ll do my best to make `vue-color` 3.0 compatible with Vue 2.7.

Closing this issue for now.
